btree/btree.c

```diff
@@ -1283,7 +1531,8 @@ static void *ff_key(const struct nd *node, int idx)
 	struct ff_head *h    = ff_data(node);
 	void           *area = h + 1;
 
-	M0_PRE(0 <= idx && idx < h->ff_used);
+	M0_PRE(ergo(!(h->ff_used == 0 && idx == 0),
+		   (0 <= idx && idx <= h->ff_used)));
```

Comment:
It's better to introduce ff_invariant(). Add ->nt_invariant() to node_type, call it from node_invariant() and call node_invariant() on entry to every node_*() function.